### PR TITLE
doc: Use FeeModes doc helper in estimatesmartfee

### PR DESCRIPTION
**Pull request by maflcko**
Not sure why this doesn't use the doc helper, probably an oversight?

### CODE REVIEW

**Comment by MarcoFalke** [comment]
Rendered diff:

```diff
diff --git a/estimatesmartfee b/estimatesmartfee
index b424d58..c3a546f 100644
--- a/estimatesmartfee
+++ b/estimatesmartfee
@@ -1,35 +1,35 @@
 estimatesmartfee conf_target ( "estimate_mode" )
 
 Estimates the approximate fee per kilobyte needed for a transaction to begin
 confirmation within conf_target blocks if possible and return the number of blocks
 for which the estimate is valid. Uses virtual transaction size as defined
 in BIP 141 (witness data is discounted).
 
 Arguments:
 1. conf_target      (numeric, required) Confirmation target in blocks (1 - 1008)
 2. estimate_mode    (string, optional, default=CONSERVATIVE) The fee estimate mode.
                     Whether to return a more conservative estimate which also satisfies
                     a longer history. A conservative estimate potentially returns a
                     higher feerate and is more likely to be sufficient for the desired
                     target, but is not as responsive to short term drops in the
                     prevailing fee market.  Must be one of:
-                    "UNSET"
-                    "ECONOMICAL"
-                    "CONSERVATIVE"
+                    "unset"
+                    "economical"
+                    "conservative"
 
 Result:
 {                   (json object)
   "feerate" : n,    (numeric, optional) estimate fee rate in BTC/kB (only present if no errors were encountered)
   "errors" : [      (json array, optional) Errors encountered during processing (if there are any)
     "str",          (string) error
     ...
   ],
   "blocks" : n      (numeric) block number where estimate was found
                     The request target will be clamped between 2 and the highest target
                     fee estimation is able to return based on how long it has been running.
                     An error is returned if not enough transactions and blocks
                     have been observed to make an estimate for any number of blocks.
 }
 
 Examples:
 > bitcoin-cli estimatesmartfee 6

File: src/rpc/mining.cpp [hunk 1]
```diff
@@ -1034,9 +1034,7 @@ static RPCHelpMan estimatesmartfee()
             "                   higher feerate and is more likely to be sufficient for the desired\n"
             "                   target, but is not as responsive to short term drops in the\n"
             "                   prevailing fee market.  Must be one of:\n"
-            "       \"UNSET\"\n"
-            "       \"ECONOMICAL\"\n"
-            "       \"CONSERVATIVE\""},
+             "\"" + FeeModes("\"\n\"") + "\""},
```
Comment (laanwj) [inline src/rpc/mining.cpp:1037]: This changes the strings to lowercase. I'm not sure that affects anything, but in any case you'd likely also want to change the CONSERVATIVE above to that.

Comment (MarcoFalke) [reply src/rpc/mining.cpp:1037]: Thanks, done

**Review by luke-jr — APPROVED** [review]
utACK

File: src/rpc/mining.cpp [hunk 2]
```diff
@@ -1022,21 +1022,19 @@ static RPCHelpMan submitheader()
...
+            {"estimate_mode", RPCArg::Type::STR, /* default */ "conservative", "The fee estimate mode.\n"
             "                   Whether to return a more conservative estimate which also satisfies\n"
             "                   a longer history. A conservative estimate potentially returns a\n"
             "                   higher feerate and is more likely to be sufficient for the desired\n"
             "                   target, but is not as responsive to short term drops in the\n"
             "                   prevailing fee market.  Must be one of:\n"
-            "       \"UNSET\"\n"
-            "       \"ECONOMICAL\"\n"
-            "       \"CONSERVATIVE\""},
+             "\"" + FeeModes("\"\n\"") + "\""},
```
Comment (jonatack) [inline src/rpc/mining.cpp:1037]: As the values will be displayed in lowercase, perhaps update the functional test:
```diff
--- a/test/functional/rpc_estimatefee.py
+++ b/test/functional/rpc_estimatefee.py
@@ -41,6 +41,8 @@ class EstimateFeeTest(BitcoinTestFramework):
         self.nodes[0].estimatesmartfee(1)
         # self.nodes[0].estimatesmartfee(1, None)
         self.nodes[0].estimatesmartfee(1, 'ECONOMICAL')
+        self.nodes[0].estimatesmartfee(1, 'unset')
+        self.nodes[0].estimatesmartfee(1, 'conservative') 
```

**Review by jonatack — COMMENTED** [review]
Tested ACK fa09513fb609dbb0d

File: src/rpc/mining.cpp [hunk 3]
```diff
@@ -1022,21 +1022,19 @@ static RPCHelpMan submitheader()
...
+        "for which the estimate is valid. Uses virtual transaction size as defined\n"
+        "in BIP 141 (witness data is discounted).\n",
+        {
+            {"conf_target", RPCArg::Type::NUM, RPCArg::Optional::NO, "Confirmation target in blocks (1 - 1008)"},
+            {"estimate_mode", RPCArg::Type::STR, /* default */ "conservative", "The fee estimate mode.\n"
             "                   Whether to return a more conservative estimate which also satisfies\n"
             "                   a longer history. A conservative estimate potentially returns a\n"
             "                   higher feerate and is more likely to be sufficient for the desired\n"
             "                   target, but is not as responsive to short term drops in the\n"
             "                   prevailing fee market.  Must be one of:\n"
```
Comment (jonatack) [inline src/rpc/mining.cpp:1036]: for info, the other RPC `estimate_mode` helps state
```suggestion
            "                   prevailing fee market.  Must be one of (case insensitive):\n"
```

**Review by kristapsk — APPROVED** [review]
ACK fa09513fb609dbb0d7381289505de72257902edc

File: src/rpc/mining.cpp (hunk 2 above)
Comment (MarcoFalke) [reply src/rpc/mining.cpp:1037]: thanks, fixed

**Comment by laanwj** [comment]
Code review ACK fa8abdc9953e381715493b259908e246914793b0

**Pull request merged**